python/tank/platform/bundle.py

```diff
@@ -394,6 +382,27 @@ def import_module(self, module_name):
         
         return sys.modules[mod_name]
 
+    def get_project_cache_location(self, project_id):
+        """
+        Gets the bundle's cache-location path for the given project id.
+
+        :param project_id:  The project Entity id number.
+
+        :returns:           str
+        """
+        # this method is memoized for performance since it is being called a lot!
+        if self.__cache_location.get(project_id) is None:
+            pc_id = self.__tk.pipeline_configuration.get_shotgun_id()
```

Comment:
Technically, I think this will function properly, but the caching mechanism won't work as intended. We wouldn't find the cached data on disk, so we would go to the database and then cache that data in the wrong place.

So, we essentially end up with multiple caches for a project depending on what pipeline config is being referenced.

It's maybe weird behavior, regardless, so we should definitely discuss.